ankihub/gui/operations/ankihub_sync.py

```diff
@@ -326,6 +330,12 @@ def _upload_if_full_sync_triggered_by_ankihub(
     on_done: Callable[[], None],
     _old: Callable[[aqt.main.AnkiQt, SyncOutput, Callable[[], None]], None],
 ) -> None:
+    if aqt.mw.col.db is None:
+        LOGGER.warning("Collection is closed, skipping custom full sync.")
+        config.set_schema_to_do_full_upload_for_once(None)
+        _old(mw, out, on_done)
```

Comment:
What about calling collection_schema() and caching its result earlier (in sync_with_ankiweb() maybe) so we don't have to skip our full sync handling?

Reply:
From what I saw in the logs I looked at, `aqt.mw.col.db` is already `None` when the sync starts.

The review heatmap add-on throws an exception caused by this before the sync starts:
```
  File "/Users/some_user/Library/Application Support/Anki2/addons21/1771074083/renderer.py", line 205, in _cache_still_valid
    col_unchanged = self._mw.col.mod == cache.col_mod  # type: ignore
  File "anki._legacy", line 81, in __getattr__
AttributeError: 'Collection' object has no attribute 'mod'
```